hail/python/hail/methods/statgen.py

```diff
@@ -364,10 +368,35 @@ def linear_regression_rows(y, x, covariates, block_size=16) -> hail.Table:
 
     cov_field_names = list(f'__cov{i}' for i in range(len(covariates)))
 
+    row_fields = dict(mt.row_key)
+    for f in pass_through:
+        if isinstance(f, str):
+            if f not in mt.row:
+                raise ValueError(f"'linear_regression_rows/pass_through': MatrixTable has no row field {repr(f)}")
+            if f in row_fields:
+                # allow silent pass through of key fields
+                if f in mt.row_key:
+                    pass
+                else:
+                    raise ValueError(f"'linear_regression_rows/pass_through': found duplicated field {repr(f)}")
+            row_fields[f] = mt[f]
+        else:
+            assert isinstance(f, Expression)
+            if not f._ir.is_nested_field:
+                raise ValueError(f"'linear_regression_rows/pass_through': expect fields or nested fields, not complex expressions")
+            if not f._indices == mt._row_indices:
```

Comment:
do we have a util to check that an expression is row indexed?

Reply:
it returns an error message that's not really appropriate here. This is pretty lightweight.